### DISCUSSION TITLE
Hidden server (master) node(s)

### DISCUSSION THREAD

**Discussion by carlocorradini** [k3s-io/k3s#6404] [Q&A] [answered]
It's possible to achieve something like managed K8s, where we hide the master (server) nodes and only show the worker (agent) nodes?
Something like described in [this](https://stackoverflow.com/questions/55727442/how-do-managed-kubernetes-providers-hide-the-master-nodes) StackOverflow question.
Thanks 🥳

**Comment by brandond** [comment, marked as answer]
It is not supported, but there is the `--disable-agent` flag that will do what you're looking for.

Note that doing so disables both containerd and the kubelet, so you cannot run any pods on these nodes or manage/upgrade them using system-upgrade-controller or the like.

**Reply by carlocorradini** [reply]
Thanks for your reply @brandond
Therefore, if I run `k3s server --disable-agent --node-taint CriticalAddonsOnly=true:NoExecute` is ok?
Do you suggest not doing this ?
Thanks again

**Reply by brandond** [reply]
If you disable the agent there is no containerd, no kubelet, and therefore no node created in the cluster... so there's nothing to taint. Like I said, you can't run any pods on it so taints are meaningless.
